### PR TITLE
Fix evaluation of fact-tools version, add command line option to add …

### PR DESCRIPTION
**Pull request by MaxNoe**
* Read Version from the jar, only possible for FACT-Tools after https://github.com/fact-project/fact-tools/pull/281

* Add possibility to provide a version string on the command-line

### CODE REVIEW

File: erna/scripts/upload.py [hunk 1]
```diff
@@ -74,17 +74,27 @@ def xml(config, fact_tools_version, name, comment, xml_file):
 
 @click.command()
 @click.option('--config', help='Yaml file containing database credentials')
+@click.option('--version', help='Set the version, if not the jar is executed to obtain it')
 @click.argument('jar-file')
-def jar(config, jar_file):
+def jar(config, version, jar_file):
```
Comment (tarrox) [inline erna/scripts/upload.py]: While we are at it add a docstring that tells us what this function is for

File: erna/scripts/upload.py [hunk 2]
```diff
@@ -74,17 +74,27 @@ def xml(config, fact_tools_version, name, comment, xml_file):
...
 
     config = load_config(config)
 
     log.debug('Connecting to database')
     database.init(**config['processing_database'])
 
-    output = sp.check_output(['java', '-jar', jar_file]).decode()
-    version = re.match('.*Version ([0-9]+\.[0-9]+\.[0-9]+)', output).groups()[0]
-    log.info('Found FACT Tools version "{}"'.format(version))
+    output = sp.check_output(['java', '-jar', jar_file, '--version']).decode()
```
Comment (tarrox) [inline erna/scripts/upload.py]: You don't need to execute this if the version is supplied so move into the if, and while we are at it you could check whether the output is empty, as versions before our version stuff don't seem to output anything

File: erna/scripts/upload.py [hunk 3]
```diff
@@ -74,17 +74,27 @@ def xml(config, fact_tools_version, name, comment, xml_file):
...
+
+    if version is None:
+        version_info = {}
+        for line in output.splitlines():
+            k, value = line.split(':')
+            version_info[k.strip()] = value.strip()
+
+        version = version_info['git description']
```
Comment (tarrox) [inline erna/scripts/upload.py]: Maybe test beforehand whether the key exists and create a nice error while failing if it doesn't exists? For example ask the user to add their own version and restart the program.

File: erna/scripts/upload.py (hunk 1 above)
Comment (MaxNoe) [reply erna/scripts/upload.py]: done

File: erna/scripts/upload.py (hunk 2 above)
Comment (MaxNoe) [reply erna/scripts/upload.py]: But this basically tests if the provided file is a valid jar, so I want to keep it above the if

**Pull request merged**